packages/sdk-js/src/plugins/third-party-tracking.ts

```diff
@@ -68,6 +68,25 @@ export function thirdPartyTrackingPlugin({
         promises.push(segmentPromise);
       }
 
+      if (trackers.includes("snowplow") && window.snowplow) {
+        window.snowplow("trackSelfDescribingEvent", {
+          event: {
+            schema: "iglu:com.growthbook/experiment_viewed/jsonschema/1-0-0",
+            data: {
+              experimentId: e.key,
+              variationId: r.key,
+              hashAttribute: r.hashAttribute,
+              hashValue: r.hashValue,
+            },
+          },
+        });
+        // snowplow uses local storage so we don't need to wait long
+        const snowplowPromise = new Promise((resolve) =>
+          window.setTimeout(resolve, 80)
```

Comment:
You can `window.snowplow(resolve)` and it will call the function once it has successfully processed the above event and committed it to localStorage.

But if the tracker hasn't loaded for some reason the Promise will never resolve (not sure what awaits this promise).